masq/src/command_processor.rs

```diff
@@ -111,17 +127,124 @@ mod tests {
+            self.sender
+                .send("; we'll do all being possible for that.".to_string())
+                .unwrap();
+            thread::sleep(Duration::from_millis(10));
+            self.sender
+                .send(" If only we have enough strength and spirit".to_string())
+                .unwrap();
+            thread::sleep(Duration::from_millis(10));
+            self.sender
+                .send(" and determination and support and... snacks.".to_string())
+                .unwrap();
+            thread::sleep(Duration::from_millis(10));
+            self.sender.send(" Roger.".to_string()).unwrap();
+            Ok(())
+        }
+    }
+
+    #[derive(Debug)]
+    struct ToUiBroadcastTrigger {}
+
+    impl Command for ToUiBroadcastTrigger {
+        fn execute(&self, context: &mut dyn CommandContext) -> Result<(), CommandError> {
+            let input = UiBroadcastTrigger {}.tmb(0);
+            context.send(input).unwrap(); //send instead of transact; using FFM.
+            Ok(())
+        }
+    }
+
+    #[test]
+    fn process_locks_writing_and_prevents_interferences_from_broadcast_messages() {
+        running_test();
+        let port = find_free_port();
+        let broadcast = UiUndeliveredFireAndForget {
+            opcode: "whateverTheOpcodeHereIs".to_string(),
+        }
+        .tmb(0);
+        let server = MockWebSocketsServer::new(port)
+            .queue_response(broadcast.clone())
+            .queue_response(broadcast.clone())
+            .queue_response(broadcast.clone())
+            .queue_response(broadcast);
+
+        let (broadcast_stream_factory, broadcast_stream_factory_handle) = TestStreamFactory::new();
+        let (cloned_stdout_sender, _) = broadcast_stream_factory.clone_senders();
+
+        let args = [
+            "masq".to_string(),
+            "--ui-port".to_string(),
+            format!("{}", port),
+        ];
+
+        let (broadcast_handle, terminal_interface) =
+            Main::populate_interactive_dependencies(broadcast_stream_factory).unwrap();
+
+        let processor_factory = CommandProcessorFactoryReal::new();
+        let stop_handle = server.start();
+        let mut processor = processor_factory
+            .make(terminal_interface, broadcast_handle, &args)
+            .unwrap();
+
+        processor
+            .process(Box::new(ToUiBroadcastTrigger {}))
+            .unwrap();
+        thread::sleep(Duration::from_millis(50));
+        processor
+            .process(Box::new(TameCommand {
+                sender: cloned_stdout_sender,
+            }))
+            .unwrap();
+
+        let tamed_message_as_a_whole = "This is a message which must be delivered as one piece; we'll do all being \
+             possible for that. If only we have enough strength and spirit and determination and support and... snacks. Roger.";
+        let received_output = broadcast_stream_factory_handle.stdout_so_far();
+        assert!(
```

Comment:
Can I see another assert here proving that the broadcast messages are also present here in this string?